### DISCUSSION TITLE
Package-diff link in PR Description is clickable but not working for private NPM packages

### DISCUSSION THREAD

**Discussion by lukaskolafa** [renovatebot/renovate#17188] [Request Help]
Hello,

I am unsure if this is a bug or if we are just missing a configuration. We are using renovate to update our NPM packages using a **private** repository in Azure DevOps. The PR description contains clickable links pointing to a diff page. Eg.:

`https://app.renovatebot.com/package-diff?name=xxxx&from=2.1.1&to=2.1.3`

The link is not working, and it shows only a spinning renovate symbol. I assume (but are not sure) that renovate does not support this feature on private packages. **Or does it?**

Anyway, if it does, how can I configure renovate so the package-diff works? And if it doesn't, then I suppose it is a bug - the version number in the PR description should not be clickable.

many thanks for your support,

best
Lukas

**Comment by viceice** [comment]
Those diff links only works for public npmjs.org packages

**Reply by lukaskolafa** [reply]
ok thx, I thought so.. I will then move this conversation into issues, where we could discuss if the links for private packages can be made non-clickable to avoid user confusion.

**Reply by viceice** [reply]
that's not easily doable, so issue would probably be closed undone

**Reply by viceice** [reply]
@rarkins do you have an idea how to solve this?

**Reply by rarkins** [reply]
It's not easy. If we only show the link for npmjs.org then those using private registries to proxy public packages would then lose their diff links

**Reply by lukaskolafa** [reply]
I see.. ok, no worries, Thanks anyway for checking it. I believe the users can live with this minor issue if they know about it. 

I tried to search if it is possible to get a list of all public npm packages - the DB is massive, but apparently, there is a way via this URL: https://replicate.npmjs.com/_all_docs  (I found it in SO here: https://stackoverflow.com/a/48272170). The list would need to be cached somewhere for fast access, and the PR creation method could check on the fly where to put the link and where not, similarly as done with the Github token for NuGet packages. But I am only guessing as I do not know the internal process of Renovate, so maybe this Idea is not feasible.

**Reply by viceice** [reply]
to complex and expensive

**Comment by gilisho** [comment]
Is there any news since the last reply in this discussion? maybe a new configuration to not make it clickable?

**Reply by gilisho** [reply]
@rarkins I assume the thumbs down reaction symbolizes a "No" answer, since I'm not sure what is wrong with me asking this question to get this reaction. Thanks.

**Reply by rarkins** [reply]
Please don't spam issues with "is there any update on this?" questions. If there were, they'd be in the issue. All it does is creates noise for maintainers. 

https://github.com/renovatebot/renovate/blob/main/docs/development/help-us-help-you.md
